lwt/examples/test_client.ml

```diff
@@ -1,25 +1,50 @@
-
 open Lwt
 open Ex_common
 
+let mypsk = ref None
+
+let ticket_cache = {
+  Tls.Config.lookup = (fun _ -> None) ;
+  ticket_granted = (fun psk epoch ->
+      Logs.info (fun m -> m "ticket granted %a %a"
+                    Sexplib.Sexp.pp_hum (Tls.Core.sexp_of_psk13 psk)
+                    Sexplib.Sexp.pp_hum (Tls.Core.sexp_of_epoch_data epoch)) ;
+      mypsk := Some (psk, epoch)) ;
+  lifetime = 0l ;
+  timestamp = Ptime_clock.now
+}
+
 let test_client _ =
+  X509_lwt.private_of_pems
+    ~cert:server_cert
+    ~priv_key:server_key >>= fun cert ->
   let port = 4433 in
   let host = "127.0.0.1" in
   X509_lwt.authenticator `No_authentication_I'M_STUPID >>= fun authenticator ->
   Tls_lwt.connect_ext
-    Tls.Config.(client ~authenticator ~ciphers:Ciphers.supported ())
+    Tls.Config.(client (* ~certificates:(`Single cert) *) ?cached_ticket:!mypsk ~ticket_cache ~authenticator ~ciphers:Ciphers.supported ())
```

Comment:
Wouldn't it be neat if the the `client` could look up the most relevant ticket in the `ticket_cache` instead of the user passing it in? (I guess the current `session_cache` is a bit geared towards a server?)

Reply:
maybe. maybe later.